core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -3198,6 +3220,227 @@ public String apply(String s) {
+                                                   SqlValidatorScope scope) {
+    SqlCall call = SqlUtil.makeCall(getOperatorTable(), identifier);
+    if (call != null) {
+      return null;
+    }
+    SqlQualified qualified = scope.fullyQualify(identifier);
+    List<String> names = qualified.identifier.names;
+
+    if (names.size() < 2) {
+      return null;
+    }
+
+    return new Pair<>(names.get(names.size() - 2), Util.last(names));
+  }
+
+  // Returns true iff the given column is valid inside the given aggCall.
+  private boolean isRolledUpColumnAllowedInAgg(SqlIdentifier identifier, SqlValidatorScope scope,
+                                               SqlCall aggCall, SqlNode parent) {
+    Pair<String, String> pair = findTableColumnPair(identifier, scope);
+
+    if (pair == null) {
+      return true;
+    }
+
+    String tableAlias = pair.left;
+    String columnName = pair.right;
+
+    Table table = findTable(tableAlias);
+    if (table != null) {
+      return table.rolledUpColumnValidInsideAgg(columnName, aggCall, parent,
+              catalogReader.getConfig());
+    }
+    return true;
+  }
+
+
+  // Returns true iff the given column is actually rolled up.
+  private boolean isRolledUpColumn(SqlIdentifier identifier, SqlValidatorScope scope) {
+    Pair<String, String> pair = findTableColumnPair(identifier, scope);
+
+    if (pair == null) {
+      return false;
+    }
+
+    String tableAlias = pair.left;
+    String columnName = pair.right;
+
+    Table table = findTable(tableAlias);
+    if (table != null) {
+      return table.isRolledUp(columnName);
+    }
+    return false;
+  }
+
+  private Table findTable(CalciteSchema schema, String tableName, boolean caseSensitive) {
+    CalciteSchema.TableEntry entry = schema.getTable(tableName, caseSensitive);
+    if (entry != null) {
+      return entry.getTable();
+    }
+
+    // Check sub schemas
+    for (CalciteSchema subSchema : schema.getSubSchemaMap().values()) {
+      Table table = findTable(subSchema, tableName, caseSensitive);
+      if (table != null) {
+        return table;
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Given a table alias, find the corresponding {@link Table} associated with it
+   * */
+  private Table findTable(String alias) {
+    List<String> names = null;
+    if (tableScope == null) {
+      // no tables to find
+      return null;
+    }
+
+    for (ScopeChild child : tableScope.children) {
+      if (catalogReader.nameMatcher().matches(child.name, alias)) {
+        names = ((SqlIdentifier) child.namespace.getNode()).names;
+        break;
+      }
+    }
+    if (names == null || names.size() == 0) {
+      return null;
+    } else if (names.size() == 1) {
+      return findTable(catalogReader.getRootSchema(), names.get(0),
+              catalogReader.nameMatcher().isCaseSensitive());
+    }
+
+    String schemaName = names.get(0);
+    String tableName = names.get(1);
```

Comment:
@zhumayun , @jcamacho, this seems to support just one schema level.
In other words, if table is located in a subschema, then table is not found.

In my case, tables are located in `schema.subschema..table`, and this code is unable to find the table.

I'm not very familiar with SqlValidator code, however the proper way to resolve tables seems to be `scope.resolveTable`

If "findTable" is a proper approach at all.

Reply:
@vlsi , I was waiting in case @zhumayun could reply... I am not that familiar with the validator code either, but I thought that follow-up lines in change deal with subschemas correctly. If this is a bug, could you create a JIRA case for this? Further, if you have a clear idea on how to solve it, e.g., via scope.resolveTable, maybe we can try to check this in before 1.14 release.